planning/obstacle_velocity_planner/src/node.cpp

```diff
@@ -0,0 +1,689 @@
+  }
+  return PlanningAlgorithm::INVALID;
+}
+
+rcl_interfaces::msg::SetParametersResult ObstacleVelocityPlannerNode::onParam(
+  const std::vector<rclcpp::Parameter> & parameters)
+{
+  planner_ptr_->updateCommonParam(parameters);
+  planner_ptr_->updateParam(parameters);
+
+  tier4_autoware_utils::updateParam<bool>(
+    parameters, "common.is_showing_debug_info", is_showing_debug_info_);
+  planner_ptr_->setParams(
+    is_showing_debug_info_, min_behavior_stop_margin_, max_nearest_dist_deviation_,
+    max_nearest_yaw_deviation_);
+
+  // obstacle_filtering
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.rough_detection_area_expand_width",
+    obstacle_filtering_param_.rough_detection_area_expand_width);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.detection_area_expand_width",
+    obstacle_filtering_param_.detection_area_expand_width);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.decimate_trajectory_step_length",
+    obstacle_filtering_param_.decimate_trajectory_step_length);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.min_obstacle_crossing_velocity",
+    obstacle_filtering_param_.min_obstacle_crossing_velocity);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.margin_for_collision_time",
+    obstacle_filtering_param_.margin_for_collision_time);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.max_ego_obj_overlap_time",
+    obstacle_filtering_param_.max_ego_obj_overlap_time);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.max_prediction_time_for_collision_check",
+    obstacle_filtering_param_.max_prediction_time_for_collision_check);
+  tier4_autoware_utils::updateParam<double>(
+    parameters, "obstacle_filtering.crossing_obstacle_traj_angle_threshold",
+    obstacle_filtering_param_.crossing_obstacle_traj_angle_threshold);
+
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = true;
+  result.reason = "success";
+  return result;
+}
+
+void ObstacleVelocityPlannerNode::onObjects(const PredictedObjects::SharedPtr msg)
+{
+  in_objects_ptr_ = msg;
+}
+
+void ObstacleVelocityPlannerNode::onOdometry(const Odometry::SharedPtr msg)
+{
+  if (current_twist_ptr_) {
+    prev_twist_ptr_ = current_twist_ptr_;
+  }
+
+  current_twist_ptr_ = std::make_unique<geometry_msgs::msg::TwistStamped>();
+  current_twist_ptr_->header = msg->header;
+  current_twist_ptr_->twist = msg->twist.twist;
+}
+
+void ObstacleVelocityPlannerNode::onSmoothedTrajectory(const Trajectory::SharedPtr msg)
+{
+  planner_ptr_->setSmoothedTrajectory(msg);
+}
+
+void ObstacleVelocityPlannerNode::onTrajectory(const Trajectory::SharedPtr msg)
+{
+  // check if subscribed variables are ready
+  if (msg->points.empty() || !current_twist_ptr_ || !prev_twist_ptr_ || !in_objects_ptr_) {
+    return;
```

Comment:
Do you have to check self_pose_listener_?
`waitForFirstPose` function seems to be written after `onTrajectory` register.

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/570/commits/8d8b4832f878598a66ef5f76abe2040935f21bf4
DONE